src/Webapi/Dom/Webapi__Dom__HtmlOptionsCollection.res

```diff
@@ -0,0 +1,31 @@
+/**
+ * Spec: https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#htmloptionscollection
+ */
+type t = Dom.htmlOptionsCollection
+
+// realistically this only deals with HtmlOption and HtmlOptGroup elements
+
+include Webapi__Dom__HtmlCollection.Impl({
+  type t = t
+})
+
+@set external setLength: (t, int) => unit = "length"
+@set_index external setItem: (t, int, Dom.htmlElement) => unit = ""
```

Comment:
I believe these functions could be refined to take `@unwrap [#Option(Dom.htmlOptionElement) | #OptGroup(Dom.htmlOptGroupElement)]`

Reply:
The spec says that it can only be one thing HTMLOptionElement and some quick testing in seems to confirm that I got this when trying to set a HTMLOptGroup.
`Failed to set an indexed property on 'HTMLOptionsCollection': parameter 2 is not of type 'HTMLOptionElement'.`
So I think this should be:
```suggestion
@set_index external setItem: (t, int, Dom.htmlOptionElement) => unit = ""
```

I think it's because it can't tell what to set the selectedIndex to at that point. The add will not alter the index unless you append before it but then it can just increase the current index.

Reply:
> Does unwrap work with anything? or only primitive types?

Looks like it, although the compiled output is a bit strange in the test

```
opts.add(({
        NAME: "Option",
        VAL: Webapi__Dom__HtmlOptionElement.ofElement(opt3)
      }).VAL, ({
        NAME: "Element",
        VAL: Webapi__Dom__HtmlElement.ofElement(opt2)
      }).VAL);
```

Reply:
oof. Fascinating. I wonder if that's a bug 🤔

I think it's fine to leave in place, since the result is correct, but it might be worth logging as a compiler issue 🤔